### ISSUE TITLE
Bug: [no-misused-spread] suggests invalid Object.fromEntries fix for WeakMap spread

### ISSUE THREAD

**Issue by cchaeyoung** [typescript-eslint/typescript-eslint#12849]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Playground Link

https://typescript-eslint.io/play/#ts=6.0.2&fileType=.tsx&code=MYewdgzgLgBA1gUwJ4wLwwN4F8DcBYAKFElgHcEBDOAWQoAc0YwFSYB1Sm%2BgCgG1fESADQwARADcKAGwCuCUQF0FASnwFCxaDABOCCDKmx0GGADpz5KrQa4gA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oFtLlZkiACa1kxaIgCGQ9FETRoHaJHBgAviDVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkAhqqrAO4BSZlemArugDTgakEl2TQ8AOW6MAwgAt00ANYcwXXoIC%2BITUA&tokens=false

### Repro Code

```TypeScript
const key = {};
const weakMap = new WeakMap([[key, "value"]]);

const result = { ...weakMap };
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  parserOptions: {
    project: true,
  },
  rules: {
    "@typescript-eslint/no-misused-spread": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "allowJs": true,
    "strictNullChecks": true
  }
}
```

### Expected Result

The rule should still report the WeakMap spread (`{ ...weakMap }`, message: `noMapSpreadInObject`), since spreading a WeakMap into an object also produces an empty object, same as with Map.

However, it should not offer the `Object.fromEntries(...)` suggestion for WeakMap, since WeakMap is not iterable. Applying that suggestion produces `Object.fromEntries(weakMap)`, which throws `TypeError: object is not iterable` at runtime.

So for this case, I'd expect the report to remain, but the `Object.fromEntries(weakMap)` suggestion to be omitted.

### Actual Result

The rule reports the WeakMap spread with `noMapSpreadInObject`, but also offers the `replaceMapSpreadInObject` suggestion ("Replace map spread in object with `Object.fromEntries()`").

The suggested replacement is `const result = Object.fromEntries(weakMap);`. This does not type-check because WeakMap is not iterable. TypeScript reports `TS2769: No overload matches this call.`

The suggested code would also throw `TypeError: object is not iterable` at runtime.

### Additional Info

This appears to come from `no-misused-spread` treating `WeakMap` together with `Map` and `ReadonlyMap` when generating the `Object.fromEntries(...)` suggestion.

The report itself seems correct, since spreading a WeakMap into an object also produces an empty object. The suggestion should likely use a narrower check that excludes WeakMap, because WeakMap is not iterable.

💖

**Comment by kirkwaiblinger** [comment]
+1. Rewording the issue concisely: by design, you can't access the contents of a WeakMap so reporting on `{ ...weakMap }` is 👍 but we should remove the suggestion, because there is no alternative to suggest (unlike a normal map).
